src/components/sidebar/sidebar.test.tsx

```diff
@@ -0,0 +1,318 @@
+    render(
+      <CarbonProvider>
+        <MockApp onCancel={() => {}} />
+      </CarbonProvider>
+    );
+
+    await user.click(screen.getByRole("button", { name: "Close" }));
+
+    await waitForElementToBeRemoved(() => screen.queryByRole("dialog"));
+    expect(document.body).toHaveFocus();
+  });
+});
+
+test("focus is trapped within sidebar when opened", async () => {
+  const user = userEvent.setup({ advanceTimers: jest.advanceTimersByTime });
+  render(
+    <CarbonProvider>
+      <Sidebar open header="My sidebar">
+        <button type="button">First</button>
+        <button type="button">Second</button>
+      </Sidebar>
+    </CarbonProvider>
+  );
+  const firstButton = screen.getByRole("button", { name: "First" });
+  firstButton.focus();
+
+  await user.tab();
+
+  expect(screen.getByRole("button", { name: "Second" })).toHaveFocus();
+
+  await user.tab();
+
+  expect(screen.getByRole("button", { name: "First" })).toHaveFocus();
+});
+
+test("focus is not trapped within sidebar when enableBackgroundUI is true", async () => {
+  const user = userEvent.setup({ advanceTimers: jest.advanceTimersByTime });
+  render(
+    <CarbonProvider>
+      <Sidebar open header="My sidebar" enableBackgroundUI>
+        <button type="button">First</button>
+        <button type="button">Second</button>
+      </Sidebar>
+    </CarbonProvider>
+  );
+  const firstButton = screen.getByRole("button", { name: "First" });
+  firstButton.focus();
+
+  await user.tab();
+
+  expect(screen.getByRole("button", { name: "Second" })).toHaveFocus();
+
+  await user.tab();
+
+  expect(firstButton).not.toHaveFocus();
+});
+
+test("can refocus sidebar container using a forwarded ref", async () => {
+  const user = userEvent.setup({ advanceTimers: jest.advanceTimersByTime });
+  const MockApp = () => {
+    const ref = React.useRef<HTMLDivElement | null>(null);
+    return (
+      <Sidebar open header="My sidebar" ref={ref}>
+        <button type="button" onClick={() => ref.current?.focus()}>
+          Refocus sidebar
+        </button>
+      </Sidebar>
+    );
+  };
+  render(
+    <CarbonProvider>
+      <MockApp />
+    </CarbonProvider>
+  );
+
+  await user.click(screen.getByRole("button", { name: "Refocus sidebar" }));
+
+  await waitFor(() => {
+    expect(screen.getByRole("dialog")).toHaveFocus();
+  });
+});
+
+test("calls a forwarded ref callback on render with sidebar element as argument", () => {
+  const ref = jest.fn();
+  render(<Sidebar open header="My sidebar" ref={ref} />);
+
+  expect(ref).toHaveBeenNthCalledWith(1, screen.getByRole("dialog"));
+});
+
+test("ensures overflowing content is scrollable", () => {
+  render(<Sidebar open />);
+
+  const sidebarContent = screen.getByTestId("sidebar-content");
+  expect(sidebarContent).toHaveStyle("overflow: auto");
+});
+
+test("does not control styling of overflowing content when there is a child Form with a sticky footer", () => {
```

Comment:
same question as above

Reply:
This test could probably be written in a more RTL way by adding a few elements and then asserting some aren't visible. We could rely on chromatic to capture the scrollbar and styling. I'm happy for this to be done in a new ticket as we have plenty of tests like this across the code base etc

Reply:
While I haven't tried it, I'd be surprised if this approach works in a simple way - I don't think RTL's `toBeVisible` takes into account whether things are in or out of the viewport. Even with Cypress and Playwright we've had to write custom utils for testing this, and that will be harder in js-dom as I'm pretty sure it makes no attempt to make things like `getBoundingClientRect` work properly.

I could be wrong though, maybe RTL has some magic baked in to do viewport checking.